File: c++/XString23.cpp
#include "pt4.h"
using namespace std;

int my_atoi(char c){
    int res = -1;
 
    //for (int i = 0; str[i] != '\0'; ++i)
    //  res = res * 10 + str[i] - '0';
	if (isdigit(c)){
		res = c - '0';
	}
    return res;
}

void Solve()
{
    Task("XString23");
	string s;
	pt >> s;
	int tmp=my_atoi(s[0]);
	if (tmp == -1)return;// pt << "error";
	int ans = tmp;
	for (int i=2;i<s.length();i+=2){
		tmp=my_atoi(s[i]);
		if (tmp == -1)return;// pt << "error";
		switch(s[i-1]){
			case '+': ans += tmp;break;
			case '-': ans -= tmp;break; 
		break;
		}

	}
	pt << ans;
	
}

File: c#/XString23.cs
// File: "XString23"
using PT4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XString23");
            string s = GetString();
            int ans = s[0]-'0';
            for (int i =1; i< s.Length-1; i+=2){
                Show(ans);
                if (s[i] == '+') ans += s[i+1]-'0';
                if (s[i] == '-') ans -= s[i+1]-'0';
                if (s[i] == '/') ans /= s[i+1]-'0';
                if (s[i] == '*') ans *= s[i+1]-'0'; 
                
            }
            Put(ans);
        }
    }
}
